### ISSUE TITLE
`ddccontrol -r 0x10` works with `-p`, but not with `/dev/i2c-4`

### ISSUE THREAD

**Issue by parke** [ddccontrol/ddccontrol#61]
If I run `ddccontrol` with `-p`, then it works.  However, if I specify the device, I get an `Invalid filename` error.  I am not using dbus.  I'm guessing the call to `ddcci_probe()` on line 313 of `main.c` has some side effect that allows the subsequent call to `ddcci_open(mon, fn, 0)` on line 357 to succeed.

I figured I would post an issue here before investigating further in case I am doing something obviously wrong.

I am running version 0.4.3 from the Ubuntu 18.04 repositories.  I can build and run a newer version, if needed.

```
$ ddccontrol -r 0x10 -p
ddccontrol version 0.4.3
Copyright 2004-2005 Oleg I. Vdovikin (oleg@cs.msu.su)
Copyright 2004-2006 Nicolas Boichat (nicolas@boichat.ch)
This program comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of this program under the terms of the GNU General Public License.

Probing for available monitors..I/O warning : failed to load external entity "/usr/share/ddccontrol-db/monitor/GSM4A98.xml"
Document not parsed successfully.
.....
Detected monitors :
 - Device: dev:/dev/i2c-4
   DDC/CI supported: Yes
   Monitor Name: VESA standard monitor
   Input type: Digital
  (Automatically selected)
Reading EDID and initializing DDC/CI at bus dev:/dev/i2c-4...
I/O warning : failed to load external entity "/usr/share/ddccontrol-db/monitor/GSM4A98.xml"
Document not parsed successfully.

EDID readings:
        Plug and Play ID: GSM4A98 [VESA standard monitor]
        Input type: Digital
=============================== WARNING ===============================
There is no support for your monitor in the database, but ddccontrol is
using a basic generic profile. Many controls will not be supported, and
some controls may not work as expected.
Please update ddccontrol-db, or, if you are already using the latest
version, please send the output of the following command to
ddccontrol-users@lists.sourceforge.net:

LANG= LC_ALL= ddccontrol -p -c -d

Thank you.
=============================== WARNING ===============================

Reading 0x10...
Control 0x10: +/50/100   [???]



$ ddccontrol -r 0x10 /dev/i2c-4
ddccontrol version 0.4.3
Copyright 2004-2005 Oleg I. Vdovikin (oleg@cs.msu.su)
Copyright 2004-2006 Nicolas Boichat (nicolas@boichat.ch)
This program comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of this program under the terms of the GNU General Public License.

Reading EDID and initializing DDC/CI at bus /dev/i2c-4...
Invalid filename (/dev/i2c-4).

DDC/CI at /dev/i2c-4 is unusable (-3).
If your graphics card need it, please check all the required kernel modules are loaded (i2c-dev, and your framebuffer driver).
```

I have two monitors:
LG L1981Q
HP 2009m (HSTND-2551-A)

The above behavior is with the LG L1981Q monitor on an Ubuntu 18.04 system.

I was trying to skip probing because the HP 2009m monitor "crashes" when receiving commands from ddccontrol.  The HP 2009m is connected to an Ubuntu 16.04 system with ddccontrol version 0.4.2.

By "crash", I mean the HP 2009m stops responding to all read or write requests from ddccontrol AND all the buttons on the HP 2009m stop working, including the power button.  The monitor continues to display live video correctly.  To turn the monitor off, then I need to remove the power cable.  I believe the HP 2009m can receive and process one write, but any second write will cause the crash.  I could be wrong about this, because due to the location of the monitor it is difficult (at present) to remove and reinsert the power cord.  So my testing thus far has been limited.

The LG L1981Q monitor does not crash.  I was just surprised that I had to probe it in order to read or write to it.  Hence this issue report.

**Comment by parke** [comment]
No, I overlooked the need for the `dev:` prefix.  Thanks!  It is working now.

Would you consider accepting one or more pull requests that do the following:

1) Improve the `ddccontrol` man page.
2) Improve the output of the `usage` function.
3) Validate `fn` in `main` and abort if `fn` does not begin with `dev:`
4) As an alternative to 3, if `fn` begins with `/dev/`, automatically prepend `dev:`

**Comment by kravemir** [comment]
> Would you consider accepting one or more pull requests that do the following:
> 1. Improve the ddccontrol man page.
> 2. Improve the output of the usage function.
> 3. Validate fn in main and abort if fn does not begin with dev:

@parke, sure, for all of these. Just go ahead. I'll be looking forward to see improvements :-)

Also, `master` branch contains unreleased D-Bus daemon. Some testing would be good (it works for me, but you know,.. bigger change, didn't want to just release it).

**Comment by kravemir** [comment]
> No, I overlooked the need for the `dev:` prefix. Thanks! It is working now.

You're welcome. I'm closing issue, therefore.

**Comment by parke** [comment]
@kravemir, Despite the fact that my base OS is Ubuntu, I don't run D-Bus at all, so it would not be easy for me to test the D-Bus daemon.

Is there a reason pull request #59 has not been accepted?

**Comment by kravemir** [comment]
> Is there a reason pull request #59 has not been accepted?

@parke, I've left it open for review (if anybody would like to). I just merged it.

> Despite the fact that my base OS is Ubuntu, I don't run D-Bus at all, so it would not be easy for me to test the D-Bus daemon.

Why you don't run D-Bus? Do you use any alternative?

**Comment by parke** [comment]
@kravemir,  I use the Openbox window manager and no desktop environment.  I turn off and disable all the "automatic" parts of Ubuntu that I can.  In 2013, I considered switching to Debian, but fonts looked better on Ubuntu.  I decided diagnosing the difference is font quality was more trouble than it would be worth, so I stayed with Ubuntu.  I am not aware of any alternatives to D-Bus, but I also don't know of any reason why I need D-Bus.
